src/ring/middleware/authorization.clj

```diff
@@ -0,0 +1,84 @@
+(ns ring.middleware.authorization
+  (:refer-clojure :exclude [update-keys])
+  (:require
+    [clojure.string :as str]
+    [ring.util.parsing :as parsing]))
+
+;; Copied from Clojure 1.11:
+(defn- update-keys
+  [m f]
+  (let [ret (persistent!
+              (reduce-kv (fn [acc k v] (assoc! acc (f k) v))
+                         (transient {})
+                         m))]
+    (with-meta ret (meta m))))
+
+(def ^{:added "0.4"} re-token68
+  "Pattern for base64/base64url/base32/base16 tokens.
+
+  See RFC 7235 Section 2 (https://datatracker.ietf.org/doc/html/rfc7235#section-2),
+  and RFC 9110 Section 11.2 (https://datatracker.ietf.org/doc/html/rfc9110#section-11.2)."
+  (re-pattern "[A-Za-z0-9._~+/-]+=*"))
+
+(def ^{:added "0.4"} re-auth-param
+  "Pattern for authentication parameters.
+
+  See RFC 7235 Section 2 (https://datatracker.ietf.org/doc/html/rfc7235#section-2),
+  and RFC 9110 Section 11.2 (https://datatracker.ietf.org/doc/html/rfc9110#section-11.2)."
+  (re-pattern (str "(" parsing/re-token ")\\s*=\\s*(?:" parsing/re-value ")")))
+
+(defn authorization
```

Comment:
This should be private, I think, unless we're 100% sure on adding it to the public API. The name could also be more like `parse-authorization`.